File: BE/lhdmir/투 포인터/Silver/1940_주몽.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        int m = Integer.parseInt(br.readLine());

        int[] materias = new int[n];
        StringTokenizer st = new StringTokenizer(br.readLine());

        for (int i = 0; i < n; i++) {
            materias[i] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(materias);

        // 왼쪽 포인터
        int left = 0;
        // 오른쪽 포인터
        int right = n - 1;

        int count = 0;

        while (left < right) {
            // 두 포인터가 가르키는 값의 합이 m과 동일할 경우
            if (materias[left] + materias[right] == m) {
                count++;
                // 왼쪽, 오른쪽 포인터를 한칸씩 이동
                left++;
                right--;
            }
            // 두 포인터가 가르키는 값의 합이 m보다 작을 경우
            else if (materias[left] + materias[right] < m) {
                left++;
            }
            // 두 포인터가 가르키는 값의 합이 m보다 클 경우
            else if (materias[left] + materias[right] > m) {
                right--;
            }
        }

        System.out.println(count);
    }
}

File: FE/Seo-garden/투포인터_슬라이딩윈도우/Silver/1940_주몽.swift
//구간합의 값이 M 보다 작으면 right 포인트를 증가시키고
//구간합의 값이 M 보다 크면 left 포인트를 증가시키고
//구간합의 값이 M 과 동일하면 count += 1 하고 구간합에서 left 값을 빼준다.
let N = Int(readLine()!)!
let M = Int(readLine()!)!
let inputArr = readLine()!.split(separator: " ").map { Int($0)! }.sorted()

var left = 0
var right = inputArr.count - 1          //배열의 마지막 값은 index - 1
var count = 0

while left < right {
    if inputArr[left] + inputArr[right] == M {
        count += 1
        left += 1
        right -= 1
        
        continue
    }
    
    if inputArr[left] + inputArr[right] < M {
        left += 1
    } else { right -= 1 }
}

print(count)
